src/fides/api/alembic/migrations/helpers/fideslang_migration_functions.py

```diff
@@ -0,0 +1,467 @@
+        )
+        bind.execute(
+            update_pr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+    # Update the Consent Request Table
+    existing_consent_requests: ResultProxy = bind.execute(
+        text("select id, preferences from consentrequest;")
+    )
+
+    for row in existing_consent_requests:
+        preferences: List[Dict] = row["preferences"]
+
+        if preferences:
+            for index, preference in enumerate(preferences):
+                preferences[index]["data_use"] = _replace_matching_data_label(
+                    data_label=preference["data_use"], data_label_map=data_use_map
+                )
+
+        update_cr_query: TextClause = text(
+            "UPDATE consentrequest SET preferences= :updated_preferences WHERE id= :id"
+        )
+        bind.execute(
+            update_cr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+
+def remove_conflicting_rule_targets(bind: Connection):
+    """
+    Iterates through all of the erasure policies and removes level 3 data categories in favor of level 2 data categories.
+
+    For example: user.demographic is preserved over user.demographic.*
+
+    This is needed because RuleTarget.create() validates all sibling rule targets to prevent invalid masking scenarios.
+    """
+    erasure_rules: ResultProxy = bind.execute(
+        text("SELECT id, key FROM rule WHERE action_type = :action_type"),
+        {"action_type": ActionType.erasure.value},
+    )
+
+    for rule in erasure_rules:
+        all_categories_query: ResultProxy = bind.execute(
+            text("SELECT data_category FROM ruletarget WHERE rule_id = :rule_id"),
+            {"rule_id": rule.id},
+        )
+        all_categories = {row.data_category for row in all_categories_query}
+
+        rule_targets = bind.execute(
+            text("SELECT id, data_category FROM ruletarget WHERE rule_id = :rule_id"),
+            {"rule_id": rule.id},
+        )
+
+        rule_targets_to_remove = []
+        for target in rule_targets:
+            parts = target.data_category.split(".")
+            if len(parts) == 3:
+                parent_category = f"{parts[0]}.{parts[1]}"
+                if parent_category in all_categories:
+                    rule_targets_to_remove.append(target)
+                    logger.info(
+                        f"Marking conflicting rule target {target.data_category} for removal from rule {rule.key}"
+                    )
+
+        if rule_targets_to_remove:
+            target_ids = [target.id for target in rule_targets_to_remove]
+            bind.execute(
+                text("DELETE FROM ruletarget WHERE id IN :target_ids"),
+                {"target_ids": tuple(target_ids)},
+            )
+            logger.info(f"Removed {len(target_ids)} conflicting rule targets")
```

Comment:
Ah right, and you're only doing this for _erasure_ rules, I was confused in testing why a conflicting access rule target was hanging around.